lib/manager/npm/post-update/yarn.ts

```diff
@@ -74,13 +74,14 @@ export async function generateLockFile(
       preCommands.push(optimizeCommand);
     }
     const commands = [];
-    let cmdOptions =
-      '--ignore-engines --ignore-platform --network-timeout 100000';
+    let cmdOptions = '';
     if (global.trustLevel !== 'high' || config.ignoreScripts) {
       if (isYarn1) {
-        cmdOptions += ' --ignore-scripts';
+        cmdOptions +=
+          ' --ignore-scripts --ignore-engines --ignore-platform --network-timeout 100000';
       } else {
         extraEnv.YARN_ENABLE_SCRIPTS = '0';
+        extraEnv.YARN_HTTP_TIMEOUT = '100000';
       }
```

Comment:
should look like:

```ts
    let cmdOptions = '';
    if (isYarn1) {
      cmdOptions += '--ignore-engines --ignore-platform --network-timeout 100000';
    } else {
      extraEnv.YARN_HTTP_TIMEOUT = '100000';
    }
    if (global.trustLevel !== 'high' || config.ignoreScripts) {
      if (isYarn1) {
        cmdOptions += ' --ignore-scripts';
      } else {
        extraEnv.YARN_ENABLE_SCRIPTS = '0';
      }
```

Reply:
Certainly! I didn't notice the if clause